mostlyai/sdk/domain.py

```diff
@@ -4017,6 +4074,110 @@ def validate_no_circular_context_references(cls, tables):
             visited.update(seen_tables)
         return tables
 
+    @model_validator(mode="after")
+    def validate_constraints(self):
+        """Validate that constraints reference existing tables and columns."""
+        if not self.constraints or not self.tables:
+            return self
+
+        table_map = {table.name: table for table in self.tables}
+        column_usage: dict[str, dict[str, int]] = {}  # table_name -> column_name -> constraint_index
+
+        for idx, constraint in enumerate(self.constraints):
+            if constraint.table_name not in table_map:
+                raise ValueError(f"table '{constraint.table_name}' referenced by constraint not found")
+
+            table = table_map[constraint.table_name]
+            table_columns = {col.name: col for col in (table.columns or [])}
```

Comment:
If we don't have `table.columns` and the user provides constraints with non-existing columns, what would be the behaviour?